src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java

```diff
@@ -2724,7 +2734,8 @@ Path createUnixGitSSH(Path key, String user, Path knownHosts) throws IOException
             w.newLine();
             w.write("fi");
             w.newLine();
-            w.write("ssh -i \"$JENKINS_GIT_SSH_KEYFILE\" -l \"$JENKINS_GIT_SSH_USERNAME\" "
+            w.write("ssh -n -T -i \"$JENKINS_GIT_SSH_KEYFILE\" -l \"$JENKINS_GIT_SSH_USERNAME\" "
+                    + "-o BatchMode=yes -o PasswordAuthentication=no -o StrictHostKeyChecking=yes "
```

Comment:
Host key checking is controller by the host key verification strategy.  We don't want to force it here.
```suggestion
                    + "-o BatchMode=yes -o PasswordAuthentication=no "
```